### DISCUSSION TITLE
Issue using DigitalInputDebounceCounter - since_last_event not updated

### DISCUSSION THREAD

**Discussion by kinyo666** [SignalK/SensESP#795] [Q&A]
I need some help to understand how the class **DigitalInputDebounceCounter** works.
https://signalk.org/SensESP/generated/docs/classsensesp_1_1_digital_input_debounce_counter.html

I use it to count the windlass revolutions with a magnetic sensor. I first tried DigitalInputCounter (without Debounce) and as expected, it counts several times the same event.
BUT when I switched to DigitalInputDebounceCounter, it doesn't work at all, even if the _ignore_interval_ms__ value is set to 1 ms.

I can't figure out where the value _since_last_event_ is updated in digital_input.cpp :
```c++
void DigitalInputDebounceCounter::handleInterrupt() {
  if (since_last_event_ > ignore_interval_ms_) {
    counter_++;
    since_last_event_ = 0;
  }
}

class DigitalInputDebounceCounter : public DigitalInputCounter {
 public:
  DigitalInputDebounceCounter(uint8_t pin, int pin_mode, int interrupt_type,
                              unsigned int read_delay,
                              unsigned int ignore_interval_ms,
                              String config_path = "")
      : DigitalInputCounter(pin, pin_mode, interrupt_type, read_delay,
                            config_path, [this]() { this->handleInterrupt(); }),
        ignore_interval_ms_{ignore_interval_ms} {}
 
 private:
  void handleInterrupt();
 
  unsigned int ignore_interval_ms_;
  elapsedMillis since_last_event_;
  virtual bool to_json(JsonObject& root) override;
  virtual bool from_json(const JsonObject& config) override;
};
``` 

I tried to find all references (from Visual Studio Code) to this variable and it didn't find any one. The value _since_last_event_ is never set or updated, so the condition in handleInterrupt() could never be true.
Is this a bug ?

Edit: I'm using SensESP v3.0.0beta2

**Comment by LuckPeerlings** [comment]
I have had some troubles to understand this as well, but the counter is updated with the line of code `since_last_event_ = 0;` The class definition is overloaded such that this assignment leads to a reset of the timer.

The main problem why it isn't working seems to be that the counter is not reacting to the loop at all. See #816 . I have adjusted the code as describe in the issue and it seems to work now. Maybe you could give it a try as well.
